sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -377,6 +378,17 @@ abstract class JdbcDialect extends Serializable with Logging {
   }
 
   private[jdbc] class JDBCSQLBuilder extends V2ExpressionSQLBuilder {
+    def inputToCaseWhenSQL(input: Expression): String =
+      "CASE WHEN " + inputToSQL(input) + " THEN 1 ELSE 0 END"
```

Comment:
shall we do recursion here? In case the CASE WHEN is nested.

Reply:
no, this is just a utility function. recursion is done anyways when going bottom-up.

Reply:
where is it? Do you mean `inputToSQL` will recursively handle CASE WHEN?

Reply:
it is done by the system/compiler. I mean, you will go recursively in those CASE WHENs anyway, no need to make it more recursive.

Reply:
ok `inputToSQL` calls `build`, so it's already recursive.